test/uk/gov/hmrc/plasticpackagingtax/registration/views/CheckLiabilityDetailsAnswersViewSpec.scala

```diff
@@ -32,141 +32,183 @@ import java.time.format.DateTimeFormatter
+        val populatedExceedIn30Row =
+          SummaryRowDetail("checkLiabilityDetailsAnswers.future.exceed",
+                           messages("general.true"),
+                           routes.LiabilityStartDateController.displayPage()
+          )
+
+        val populatedIsLiableRow =
+          SummaryRowDetail("checkLiabilityDetailsAnswers.future.liable",
+                           messages("general.true"),
+                           routes.LiabilityStartDateController.displayPage()
+          )
+
+        "registration populated" in {
+          assertSummaryRows(preLaunchView,
+                            List(populatedWeightRow, populatedExceedIn30Row, populatedIsLiableRow)
+          )
+        }
+
+        "registration unpopulated" in {
+          val unpopulatedIsLiableRow = populatedIsLiableRow.copy(value = "")
+
+          assertSummaryRows(preLaunchViewWithEmptyRegistration,
+                            List(unpopulatedWeightRow, unpopulatedIsLiableRow)
+          )
+        }
+      }
 
-      viewWithEmptyRegistration.getElementsByClass("govuk-summary-list__key").get(
-        1
-      ) must containMessage("checkLiabilityDetailsAnswers.date")
-      viewWithEmptyRegistration.getElementsByClass("govuk-summary-list__value").get(
-        1
-      ).text() mustBe ""
-    }
+      "'preLaunch' feature flag not enabled" when {
+        val populatedLiableDateRow = SummaryRowDetail("checkLiabilityDetailsAnswers.date",
+                                                      populatedRegistration.liabilityDetails.startDate.get.asLocalDate.format(
+                                                        DateTimeFormatter.ofPattern("dd MMM yyyy")
+                                                      ),
+                                                      routes.LiabilityStartDateController.displayPage()
+        )
 
-    val summaryList = view.getElementsByClass("govuk-summary-list")
+        "registration populated" in {
+          assertSummaryRows(postLaunchView, List(populatedWeightRow, populatedLiableDateRow))
+        }
 
-    "display link to change liability date" in {
+        "registration unpopulated" in {
+          val unpopulatedLiableDateRow = populatedLiableDateRow.copy(value = "")
 
-      summaryList.first.getElementsByClass("govuk-link").get(1) must haveHref(
-        routes.LiabilityStartDateController.displayPage()
-      )
+          assertSummaryRows(postLaunchViewWithEmptyRegistration,
+                            List(unpopulatedWeightRow, unpopulatedLiableDateRow)
+          )
+        }
+      }
     }
 
-    "display liability weight" in {
-
-      view.getElementsByClass("govuk-summary-list__key").first() must containMessage(
-        "checkLiabilityDetailsAnswers.weight"
-      )
-      view.getElementsByClass("govuk-summary-list__value").first() must containText(
-        registration.liabilityDetails.weight.get.totalKg.get.toString
-      )
+    "display 'Save And Continue' button" in {
```

Comment:
Small one but this button should now read `Confirm and continue` rather than `Save and Continue`.

Reply:
Really? Is that something we've agreed to change recently?

Reply:
Is that so that we are consistent with what GRS use?

Reply:
@tmc-ee are you able to sanction this now?